File: src/main/java/com/github/thorben/webnd/domain/character/BaseValue.java
package com.github.thorben.webnd.domain.character;

import com.github.thorben.webnd.foundation.AbstractEntity;

import javax.persistence.*;

@Entity(name = "BaseValue")
@Table(name = "base_value")
public class BaseValue extends AbstractEntity<Integer> implements TypeFill {

	@Id
	@GeneratedValue
	private Integer id;
	@JoinColumn(name = "id", table = "skill_types")
	private SkillType type;
	@Column(name = "skill_value")
	private String value;
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "character_id")
	private Character character;

	protected BaseValue() {
		// JPA
	}

	public BaseValue(SkillType type, String value) {
		this.type = type;
		this.value = value;
	}

	public SkillType getType() {
		return type;
	}

	public String getValue() {
		return value;
	}

	@Override
	public Integer getId() {
		return id;
	}

}

File: src/main/kotlin/com/github/thorben/webnd/domain/character/BaseValue.kt
package com.github.thorben.webnd.domain.character

import com.github.thorben.webnd.foundation.AbstractEntity
import javax.persistence.*

@Entity(name = "BaseValue")
@Table(name = "base_value")
class BaseValue : AbstractEntity<Int?>, TypeFill {
	@Id
	@GeneratedValue
	override val id: Int? = null

	@JoinColumn(name = "id", table = "skill_types")
	override var type: SkillType? = null
		private set

	@Column(name = "skill_value")
	override var value: String? = null
		private set

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "character_id")
	private val character: Character? = null

	protected constructor() {
		// JPA
	}

	constructor(type: SkillType?, value: String?) {
		this.type = type
		this.value = value
	}

}